java_training/src/com/lftechnology/december31/GuessingGameRun.java

```diff
@@ -0,0 +1,136 @@
+package com.lftechnology.december31;
+
+import java.util.InputMismatchException;
+import java.util.LinkedList;
+import java.util.ListIterator;
+import java.util.Random;
+import java.util.Scanner;
+
+/**
+ * Guessing a number game application: User will select the guessing value from
+ * the keyboard and application will generate a random no between 1 and 100.
+ * Checks if the guess value matched with random no. or not, if matched shows
+ * the guessing value, random value and the no. of attempts.
+ * 
+ * @author nerisa
+ * 
+ */
+public class GuessingGameRun {
+	public static void main(String[] args) {
+		GuessingGame game = new GuessingGame(1, 100);
+		game.play();
+	}
+}
+
+/**
+ * Controls the game play
+ * 
+ * @author nerisa
+ * 
+ */
+class GuessingGame {
+	private int userEnteredNumber = 0;
+	private LinkedList<Integer> userGuesses = new LinkedList<Integer>();
```

Comment:
Why LinkedList , Why not arraylist . Arraylist is faster than linked list .Check it and always try to instantiate class from its parent class if it has. So here you can do private List<Integer> userGuesses = new LinkedList<Integer>();